yesod-core/src/Yesod/Core/Types.hs

```diff
@@ -306,6 +308,24 @@ data Content = ContentBuilder !BB.Builder !(Maybe Int) -- ^ The content and opti
              | ContentFile !FilePath !(Maybe FilePart)
              | ContentDontEvaluate !Content
 
+-- | Represents a Content as a String, rendering at most a specified number of
+-- bytes of the content, and annotating it with the remaining length.
+--
+-- @since 1.6.28.0
+contentToTruncatedString :: Content -> I.Int64 -> String
+contentToTruncatedString (ContentBuilder builder maybeLength) maxLength =
+    let
+      truncated = (T.unpack . Data.Text.Encoding.decodeUtf8) $ L.toStrict $ L.take maxLength (BB.toLazyByteString builder)
```

Comment:
style nit: it's nice to have a consistent flow of function application. You can do almost all `.` with a single `$`, or all `$`:

```suggestion
      truncated = T.unpack $ Data.Text.Encoding.decodeUtf8 $ L.toStrict $ L.take maxLength $ BB.toLazyByteString builder
```


```suggestion
      truncated = T.unpack . Data.Text.Encoding.decodeUtf8 . L.toStrict . L.take maxLength $ BB.toLazyByteString builder
```